src/main/java/datawave/microservice/query/messaging/kafka/KafkaQueryResultsManager.java

```diff
@@ -0,0 +1,200 @@
+        createTopic(queryId);
+        return new KafkaQueryResultsListener(messagingProperties, kafkaConsumerFactory, listenerId, queryId);
+    }
+    
+    /**
+     * Create a publisher for a specific query id.
+     * 
+     * @param queryId
+     *            The query ID to publish to
+     * @return a query result publisher
+     */
+    @Override
+    public QueryResultsPublisher createPublisher(String queryId) {
+        createTopic(queryId);
+        KafkaTemplate<String,String> kafkaTemplate = new KafkaTemplate<>(kafkaProducerFactory);
+        kafkaTemplate.setDefaultTopic(queryId);
+        return new KafkaQueryResultsPublisher(kafkaTemplate);
+    }
+    
+    /**
+     * Delete a queue for a query
+     *
+     * @param queryId
+     *            the query ID
+     */
+    @Override
+    public void deleteQuery(String queryId) {
+        deleteTopic(queryId);
+    }
+    
+    private void deleteTopic(String topic) {
+        try {
+            if (adminClient.listTopics().names().get().contains(topic)) {
+                // @formatter:off
+                adminClient
+                        .deleteTopics(Collections.singleton(topic))
+                        .topicNameValues()
+                        .get(topic)
+                        .get();
+                // @formatter:on
+            }
+        } catch (InterruptedException | ExecutionException e) {
+            log.error("Failed to delete topic " + topic, e);
+        }
+    }
+    
+    private void createTopic(String topic) {
+        try {
+            if (!adminClient.listTopics().names().get().contains(topic)) {
+                // @formatter:off
+                adminClient
+                        .createTopics(Collections.singleton(TopicBuilder
+                                .name(topic)
+                                .partitions(messagingProperties.getKafka().getPartitions())
+                                .replicas(messagingProperties.getKafka().getReplicas())
+                                .build()))
+                        .values()
+                        .get(topic)
+                        .get();
+                // @formatter:on
+            }
+        } catch (InterruptedException | ExecutionException e) {
+            log.error("Failed to delete topic " + topic, e);
+        }
+    }
+    
+    @Override
+    public void emptyQuery(String name) {
+        TopicDescription topic = describeTopic(name);
+        if (topic != null) {
+            Map<TopicPartition,RecordsToDelete> partitions = new HashMap<>();
+            RecordsToDelete records = RecordsToDelete.beforeOffset(Long.MAX_VALUE);
+            for (TopicPartitionInfo info : topic.partitions()) {
+                TopicPartition partition = new TopicPartition(name, info.partition());
+                partitions.put(partition, records);
+            }
+            DeleteRecordsResult result2 = adminClient.deleteRecords(partitions);
```

Comment:
Why result2 as opposed to result? If not using result consider a different more descriptive name

Reply:
it might have made sense at one point, but it certainly doesn't now.  fixing.